internal/manifests/collector/collector.go

```diff
@@ -53,7 +55,7 @@ func Build(params manifests.Params) ([]client.Object, error) {
 		res, err := factory(params.Config, params.Log, params.Instance)
 		if err != nil {
 			return nil, err
-		} else if res != nil {
+		} else if res != nil && !reflect.ValueOf(res).IsNil() {
```

Comment:
Is there a better way to do this? I don't think so....

Reply:
Should a factory ever being nil? especially, if the `err` is not nil? :no_good: 

That could be an alternative with the same outcome:
https://go.dev/play/p/7t-9SOo9FpX

Reply:
Ill link the comment from @swiatekm-sumo later, am afk... but IIRC In this case its because the result is an interface pointer to a nil structure so we need to check that the place it points to is actually nil.

Reply:
I think i'll do your suggestion though

Reply:
https://github.com/open-telemetry/opentelemetry-operator/pull/1965#discussion_r1281705719

Reply:
unfortunately this didn't work for this case :/
![Screenshot 2023-09-13 at 11 56 56 AM](https://github.com/open-telemetry/opentelemetry-operator/assets/10070047/41dd2848-c7c2-4ecd-9dcc-123ccdd46373)

Reply:
I'm going to create this as a re-usable function that we can resolve in the future.